extensions/amp-viewer-gpay-button/0.1/amp-viewer-gpay-button.js

```diff
@@ -14,34 +14,49 @@
  * limitations under the License.
  */
 
-import {Layout} from '../../../src/layout';
+import {AmpPaymentGoogleIntegration} from '../../../src/service/payments/amp-payment-google';
+import {getServiceForDoc} from '../../../src/service';
 
-export class AmpViewerGpayButton extends AMP.BaseElement {
+/** @const {string} */
+const TAG = 'amp-viewer-gpay-button';
+
+class AmpViewerGpayButton extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+  }
 
-    /** @private {string} */
-    this.myText_ = 'hello world';
-
-    /** @private {?Element} */
-    this.container_ = null;
+  /** @override */
+  isLayoutSupported(unusedLayout) {
+    return true;
```

Comment:
Is the page author meant to control the size of this button? Or is it always the same size?

This should probably be `return isLayoutSizeDefined(layout)` which omits `nodisplay` and `container`.

Reply:
I think this should also be consistent with the values supported in the validator file, which right now is only: 

https://github.com/ampproject/amphtml/blob/05c41cd2d15b871fdb0e0882720c5a37bafdd3e5/extensions/amp-viewer-gpay-button/validator-amp-viewer-gpay-button.protoascii#L33-L35

Otherwise layouts other than responsive will be determined to be not valid AMP by the validator.

Reply:
we planned to update the validator files in the follow up cl ; we could make those changes in this cl if required